src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,702 @@
+          case "ArrowLeft":
+            this.minuteEl.focus();
+            break;
+          case "ArrowRight":
+            if (this.hourDisplayFormat === "12") {
+              this.amPmEl.focus();
+            }
+            break;
+        }
+        break;
+      case this.amPmEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            if (this.step !== 60) {
+              this.secondEl.focus();
+            } else {
+              this.minuteEl.focus();
+            }
+            break;
+        }
+        break;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(): Promise<void> {
+    this.hourEl.focus();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private amPmKeyDownHandler = (event: KeyboardEvent): void => {
+    switch (event.key) {
+      case "a":
+        this.setTime("ampm", "AM");
+        break;
+      case "p":
+        this.setTime("ampm", "PM");
+        break;
+      case "Backspace":
+        this.setTime("ampm", "--");
+        break;
+      case "ArrowUp":
+        event.preventDefault();
+        this.incrementAmPm();
+        break;
+      case "ArrowDown":
+        event.preventDefault();
+        this.decrementAmPm();
+        break;
+    }
+  };
+
+  private decrementAmPm = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") return;
+    switch (this.ampm) {
+      case "--":
+      case "AM":
+        this.setTime("ampm", "PM");
+        break;
+      case "PM":
+        this.setTime("ampm", "AM");
+        break;
+    }
+  };
+
+  private decrementHour = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") return;
+    switch (this.hour) {
+      case "--":
+        this.setTime("hour", 0);
+        break;
+      case "00":
+        this.setTime("hour", 23);
+        break;
+      default:
+        this.setTime("hour", parseInt(this.hour) - 1);
+        break;
+    }
+  };
+
+  private decrementMinuteOrSecond = (key: MinuteOrSecond): void => {
+    if (this[key] === "--") {
+      this.setTime(key, 59);
+    } else {
+      const valueAsNumber = parseInt(this[key]);
+      if (valueAsNumber === 0) {
+        this.setTime(key, 59);
+      } else {
+        this.setTime(key, valueAsNumber - 1);
+      }
+    }
+  };
+
+  private decrementMinute = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") return;
+    this.decrementMinuteOrSecond("minute");
+  };
+
+  private decrementSecond = (event?: KeyboardEvent | MouseEvent): void => {
```

Comment:
I think it might be cleaner to add separate event handlers that call the decrementX method.